[EDU-1960] Add architecture articles to the platform section (#2623)
ably/docs: src/pages/docs/platform/architecture/access-network.mdx

---

> title: Access network

Author: b77ec120095617b99a5b43fb61dc82731224f02b

meta_description: "Understand how Ably's access network provides reliable, low-latency connectivity to its platform from anywhere in the world."
---

The access network serves as the entry point to Ably's realtime infrastructure and is core to the platform's ability to deliver reliable, low-latency messaging services globally.

## Why the access network matters

The access network must ensure that clients can reliably connect to the closest available datacenter, providing optimal performance whilst maintaining resilience against various types of failures. Given that Ably is mission-critical for many applications, the reliability of the access network is essential; any connectivity issues can potentially impact thousands or millions of end users, making the design and operation of the access network a key element of Ably's infrastructure.

## Access network architecture

Ably's access network is designed to provide reliable, low-latency connectivity to the Ably platform from anywhere in the world. The architecture leverages multiple AWS services and custom infrastructure components to ensure resilience, performance, and security.

### Global deployment across multiple regions

> Ably operates in multiple regions around the world, with servers distributed across more than 15 physical datacenters within Amazon's cloud infrastructure. This global distribution ensures that clients can connect to a nearby datacenter, minimizing the latency between the client and the entry point to Ably's network.

Reviewer: ```suggestion
Ably operates in multiple regions around the world, with servers distributed across more than 15 physical datacenters within the AWS network. This global distribution ensures that clients can connect to a nearby datacenter, minimizing the latency between the client and the entry point to Ably's network.
```

Author: (commit references in Matt's comments above - thanks for providing the info)

> Ably operates in multiple regions around the world, with servers distributed across more than 15 physical datacenters within Amazon's cloud infrastructure. This global distribution ensures that clients can connect to a nearby datacenter, minimizing the latency between the client and the entry point to Ably's network.

Author: b77ec120095617b99a5b43fb61dc82731224f02b

Each datacenter operates independently, with its own dedicated infrastructure for handling client connections, message processing, and storage. This isolation ensures that issues affecting one datacenter do not propagate to others, maintaining service availability even during regional outages.

The global distribution of datacenters also enables Ably to comply with data residency requirements by keeping data within specific geographic regions when required. This is important for customers operating in regulated industries or jurisdictions with strict data sovereignty laws.

### AWS CloudFront and network load balancers

Client connections to Ably are handled through a combination of AWS CloudFront for global edge distribution and AWS EC2 Network Load Balancers (NLBs) for traffic routing within each region.

> AWS CloudFront serves as the primary entry point to Ably's network, with over 175 edge locations globally. When a client attempts to connect to Ably, the request is first routed to the nearest CloudFront edge location. This reduces the public internet transit time, as clients connect to a nearby edge node rather than traversing the entire distance to an Ably datacenter.

Reviewer: ```suggestion
AWS CloudFront serves as the primary entry point to Ably's network, with over 700 edge locations globally. When a client attempts to connect to Ably, the request is first routed to the nearest CloudFront edge location. This reduces the public internet transit time, as clients connect to a nearby edge node rather than traversing the entire distance to an Ably datacenter.
```

> AWS CloudFront serves as the primary entry point to Ably's network, with over 175 edge locations globally. When a client attempts to connect to Ably, the request is first routed to the nearest CloudFront edge location. This reduces the public internet transit time, as clients connect to a nearby edge node rather than traversing the entire distance to an Ably datacenter.

Author: b77ec120095617b99a5b43fb61dc82731224f02b

## Protection against denial of service attacks

As a critical infrastructure provider for many businesses, Ably is a potential target for denial of service (DoS) attacks. The access network includes multiple layers of protection to detect, mitigate, and absorb such attacks without impacting legitimate traffic.

### Layer 3/4 DoS protection with AWS Shield Advanced

Ably utilizes AWS Shield Advanced to protect against volumetric DDoS attacks at the network and transport layers (Layers 3 and 4). These attacks typically involve flooding the target with a high volume of packets or connection attempts, aiming to exhaust network bandwidth or connection handling capacity.

AWS Shield Advanced provides automatic attack detection by continuously monitoring traffic patterns to detect anomalies that indicate an attack is underway. When an attack is detected, Shield automatically applies mitigations to filter malicious traffic while allowing legitimate traffic to pass through.

> During large-scale attacks, AWS can leverage its global network infrastructure to absorb and diffuse attack traffic, preventing it from reaching Ably's resources. AWS Shield includes access to the AWS DDoS Response Team (DRT), providing expert assistance during attacks, and offers financial safeguards against the increased costs that can result from DDoS attacks, including data transfer fees and scaling costs.

Author: 2ea86e786d0506d6018c371c89550a9c886e5abb

The connection handling infrastructure includes mechanisms to detect and mitigate connection-based attacks, such as slow loris attacks or connection flooding. In extreme cases, Ably can implement graceful load shedding, temporarily rejecting new connections or non-critical requests to preserve core functionality for existing connections.

This scalable infrastructure ensures that even if attack traffic reaches Ably's application servers, the impact on legitimate users is minimized. By rapidly scaling to absorb traffic spikes and implementing resource controls at multiple levels, Ably maintains service availability even during significant attack events.

## Global connectivity considerations

Operating a truly global access network presents unique challenges and considerations that Ably has addressed in its architecture.

### Geographic coverage and edge presence

Ably's access network is designed to provide low-latency connectivity from virtually anywhere in the world. The combination of AWS CloudFront's extensive edge network and Ably's strategically located datacenters ensures that most clients can connect to an entry point with minimal network latency.

> The geographic distribution of these access points is continuously reviewed and optimized based on traffic patterns and customer requirements. As Ably's user base expands into new regions, additional edge locations and datacenters can be added to maintain optimal performance.

Reviewer: ```suggestion
The geographic distribution of these access points is continuously reviewed and optimized based on traffic patterns and customer requirements. As Ably's user base expands into new regions, additional datacenters can be added to maintain optimal performance.
```

> The geographic distribution of these access points is continuously reviewed and optimized based on traffic patterns and customer requirements. As Ably's user base expands into new regions, additional edge locations and datacenters can be added to maintain optimal performance.

Author: b77ec120095617b99a5b43fb61dc82731224f02b
